components/autocomplete/Autocomplete.js

```diff
@@ -0,0 +1,144 @@
+
+const Autocomplete = ({
+    list,
+    selectedItems,
+    inputValue,
+    onRemove,
+    onSubmit,
+    onSelect,
+    onInputValueChange,
+    onOpen,
+    onClose,
+    onHighlight,
+    ...rest
+}) => {
+    let [awesomplete, setAwesomplete] = useState(null);
+    const inputRef = useRef(null);
+    const containerRef = useRef(null);
+
+    const handleInputValueChange = ({ target }) => {
+        onInputValueChange(target.value);
+    };
+
+    const handleSubmit = (item = { label: inputValue, value: inputValue, invalid: true }) => {
+        if (item.value) {
+            onSubmit(item);
+        }
+    };
+
+    const handleSelect = ({ text }) => {
+        handleSubmit(text);
+        onSelect(text);
+    };
+
+    const handleKeyDown = (e) => {
+        if (e.key === 'Backspace' && !inputValue) {
+            onRemove(selectedItems.length - 1);
+        }
+    };
+
+    useEffect(() => {
+        const awesompleteInstance = new Awesomplete(inputRef.current, {
+            container: () => containerRef.current,
+            ...rest
+        });
+
+        setAwesomplete(awesompleteInstance);
+
+        inputRef.current.addEventListener('awesomplete-selectcomplete', handleSelect);
+        inputRef.current.addEventListener('awesomplete-close', onClose);
+        inputRef.current.addEventListener('awesomplete-highlight', onHighlight);
+        inputRef.current.addEventListener('awesomplete-open', onOpen);
+
+        return () => {
+            awesompleteInstance.destroy();
+            inputRef.current.removeEventListener('awesomplete-selectcomplete', handleSelect);
+            inputRef.current.removeEventListener('awesomplete-close', onClose);
+            inputRef.current.removeEventListener('awesomplete-highlight', onHighlight);
+            inputRef.current.removeEventListener('awesomplete-open', onOpen);
+        };
+    }, []);
+
+    useEffect(() => {
+        if (awesomplete) {
+            awesomplete.list = list;
+            awesomplete.evaluate();
+        }
+    }, [awesomplete, list]);
+
+    return (
+        <form
+            className="autocomplete awesomplete"
+            onSubmit={(e) => {
+                e.preventDefault();
+                handleSubmit();
+            }}
+        >
+            <div className="autocomplete-container" ref={containerRef}>
+                <div className="flex pm-field">
+                    {selectedItems.map((item, i) => (
+                        <SelectedItem key={i} item={item} onRemove={() => onRemove(i)} />
+                    ))}
+
+                    <input
+                        value={inputValue}
+                        className="w100 autocomplete-input"
+                        spellCheck={false}
+                        autoComplete="off"
+                        autoCapitalize="off"
+                        onChange={handleInputValueChange}
+                        ref={inputRef}
+                        onBlur={() => handleSubmit()}
```

Comment:
No need to add the extra function here :)

Reply:
Yeah, gonna write one handleSubmit and reuse them between form and blur.